mergify_engine/context.py

```diff
@@ -745,31 +745,58 @@ async def _get_consolidated_data(self, name):
             return [r["user"]["login"] for r in comments if r["state"] == "COMMENTED"]
 
         # NOTE(jd) The Check API set conclusion to None for pending.
-        # NOTE(sileht): "pending" statuses are not really trackable, we
-        # voluntary drop this event because CIs just sent they status every
-        # minutes until the CI pass (at least Travis and Circle CI does
-        # that). This was causing a big load on Mergify for nothing useful
-        # tracked, and on big projects it can reach the rate limit very
-        # quickly.
-        # NOTE(sileht): Not handled for now: cancelled, timed_out, or action_required
         elif name in ("status-success", "check-success"):
             return [
                 ctxt
                 for ctxt, state in (await self.checks).items()
                 if state == "success"
             ]
         elif name in ("status-failure", "check-failure"):
+            # hopefully "cancelled" is actually a failure state to github.
+            # I think it is, however it could be the same thing as the
+            # "skipped" status.
             return [
                 ctxt
                 for ctxt, state in (await self.checks).items()
                 if state == "failure"
+                or state == "cancelled"
+                or state == "timed_out"
+                or state == "action_required"
             ]
         elif name in ("status-neutral", "check-neutral"):
             return [
                 ctxt
                 for ctxt, state in (await self.checks).items()
                 if state == "neutral"
             ]
+        elif name in ("status-skipped", "check-skipped"):
+            # hopefully this handles the gray "skipped" state that github actions
+            # workflows can send when a job that depends on a job and the job it
+            # depends on fails, making it get skipped automatically then.
+            return [
+                ctext
+                for ctext, state in (await self.checks).items()
+                if state == "skipped"
+            ]
+        elif name in ("status-pending", "check-pending"):
```

Comment:
`yellow dot `in GitHub UI is, unfortunately, more complicated, GitHub have cases where you see it, but when you query the check-runs and statuses API, there are no checks at all.

The two cases we have seen:
* context configured in GitHub branch protection, but the third-party CI haven't yet reported any status
* GitHub action has started, but check-runs API is up2date only when the container effectively started.

Our plan was to also add all contexts present in branch protection and not yet in the `self.checks` list in the pending list.
For the second point, I think we can ignore it, as it's only annoying for a short time.

Reply:
Hmm how would you be able to get those present in branch protection, but not in the ``self.checks`` list?

Reply:
@sileht how can one get that context configured in the branch protection (like how mergify can tell you specific rules on the summary when it sees that branch protection is set for specific checks)?

Reply:
You have an example here: https://github.com/Mergifyio/mergify-engine/blob/a536625582a76cd733981ea6155c1db6c14bd52b/mergify_engine/actions/merge_base.py#L153-L165

Reply:
Oh I see you get the branch then you check the response headers.